client/handle_http.go

```diff
@@ -2122,31 +2126,52 @@ func runPut(request *http.Request, responseChan chan<- *http.Response, errorChan
 }
 
 // Walk a remote directory in a WebDAV server, emitting the files discovered
-func (te *TransferEngine) walkDirDownload(job *clientTransferJob, transfers []transferAttemptDetails, files chan *clientTransferFile, url *url.URL) error {
-	// Create the client to walk the filesystem
-	rootUrl := *url
-	if job.job.namespace.DirListHost != "" {
-		// Parse the dir list host
-		dirListURL, err := url.Parse(job.job.namespace.DirListHost)
+func (te *TransferEngine) walkDirDownload(job *clientTransferJob, transfers []transferAttemptDetails, files chan *clientTransferFile, remoteURL *url.URL) error {
+	// Query the director a PROPFIND to see if we can get our directory listing
+	resp, err := queryDirector("PROPFIND", remoteURL.Path, job.job.directorUrl+"/api/v1.0/director/origin")
```

Comment:
Can you construct the `job.job.directorUrl` with `url.JoinPath` to make sure it's valid? Or are we otherwise sure this construction won't result in a double `/`?